cpp/src/reductions/reduction_prod.cu

```diff
@@ -0,0 +1,27 @@
+/*
```

Comment:
None of these files needs to be prefixed with `reduction`, and then the abbreviation can be removed. So instead of `reduction_prod.cu` rename to `product.cu`. Likewise with all the other files.

Reply:
Done